tests/unit/test_hardware_parallelism.py

```
@@ -0,0 +1,291 @@
+
+# ─────────────────────────────────────────────────────────────────────────────
+# config_helpers: multi-AMD unit registration
+# ─────────────────────────────────────────────────────────────────────────────
+
+
+class TestAppendAmdUnits:
+    """append_amd_units must register up to max_amd units."""
+
+    def test_single_amd_unit(self):
+        """Appends single amd:0 unit when max_amd is 1."""
+        append_amd_units_fn = getattr(ch, "_append_amd_units")
```

Ruff (0.16.1)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)